packages/flutter/lib/src/rendering/editable_line.dart

```diff
@@ -111,6 +122,31 @@ class RenderEditableLine extends RenderBox {
     markNeedsPaint();
   }
 
+  List<TextSelectionPoint> getEndpointsForSelection(TextSelection selection) {
+    _textPainter.layout(); // TODO(mpcomplete): is this hacky?
```

Comment:
Yeah, we need to re-think how we handle the text painter's layout more generally.  Not something we need to solve in this patch.